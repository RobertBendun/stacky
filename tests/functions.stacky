# dot compare
add2 fun # (n -- 2 + n)
	2 +
end

2 add2 .

factorial fun # (n -- n!)
	dup 1 > if
		dup 1 - factorial *
	else
		drop 1
	end
end

5 factorial .

&factorial &factorial = .

5 &factorial call .
