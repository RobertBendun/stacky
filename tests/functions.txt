4
120
1
