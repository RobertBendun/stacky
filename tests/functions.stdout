4
120
1
120
